lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -81,6 +81,9 @@ module Cardano.Wallet.Write.Tx
     , modifyTxOutputs
     , modifyLedgerBody
     , emptyTx
+    , Babbage.totalCollateralBabbageTxBodyL
+    , Babbage.collateralReturnBabbageTxBodyL
+    , Babbage.collateralInputsBabbageTxBodyL
```

Comment:
These require the `Babbage.BabbageEraTxBody (ShelleyLedgerEra era)` constraint, which we currently would need to get from `withConstraints`.

Really, we should either redefine these to rely on `IsRecentEra` instead, OR, have `IsRecentEra` imply `BabbageEraTxBody`. In either case, we need to get rid of `ShelleyLedgerEra cardanoEra` and use ledger era types directly.

Reply:
Dropping `BabbageTxBody` from the names would be nice though.

Reply:
> Dropping BabbageTxBody from the names would be nice

For a later PR though.